homeassistant/components/freebox/__init__.py

```diff
@@ -57,23 +75,15 @@ async def async_reboot(call: ServiceCall) -> None:
 
     hass.services.async_register(DOMAIN, SERVICE_REBOOT, async_reboot)
 
-    async def async_close_connection(event):
-        """Close Freebox connection on HA Stop."""
-        await router.close()
-
-    entry.async_on_unload(
-        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, async_close_connection)
-    )
-
     return True
 
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
     if unload_ok:
-        router = hass.data[DOMAIN].pop(entry.unique_id)
-        await router.close()
+        router: FreeboxRouter = hass.data[DOMAIN].pop(entry.unique_id)
+        await router._api.close()  # pylint: disable=[protected-access]
         hass.services.async_remove(DOMAIN, SERVICE_REBOOT)
```

Comment:
This is how to remove services during unload:
https://github.com/home-assistant/core/blob/f82183f0e3409edb0815476055202d39aafd645c/homeassistant/components/simplisafe/__init__.py#L460-L469

We take more than one config entry into consideration in the service handler though, as there's no way to identify which router to target in the service schema. So this service currently doesn't work with more than one config entry.

Reply:
I agree - I think it should be deprecated and moved to a button entity.
I think it should kept as it is, and dealt with in a follow-up PR.